pennylane/measurements/measurements.py

```diff
@@ -364,11 +364,9 @@ def has_decomposition(self):
         return False if self.obs is None else self.obs.has_diagonalizing_gates
 
     @property
-    def samples_computational_basis(self):
-        r"""Bool: Whether or not the MeasurementProcess returns samples in the computational basis or counts of
-        computational basis states.
-        """
-        return False
+    def measures_computational_basis(self):
+        r"""Bool: Whether or not the MeasurementProcess measures in the computational basis."""
+        return self.obs is None
```

Comment:
```suggestion
        if self.obs is None:
            return True
       try:
           if len(self.obs.diagonalizing_gates()) == 0:
               return True
      except qml.operation.DiagonalizingGatesUndefinedError:
          return False
     return False
```
Indentation is probably wrong.

Motivating example:
```
>>> qml.expval(qml.PauliZ(0)).computational_basis
False
```

Reply:
But I guess we don't particularly want to do this for op arithmetic classes, since that could be expensive.  Let me think about this...

Reply:
I think at some point we should merge the behaviour of wires and of PauliZ observables. Possibly always convert the PauliZ to wires in the observables.

Reply:
Good call! What about just checking if `self.obs` is a pauli word containing only `Z` operations? You can check the implementation in [the PR targetting the master branch](https://github.com/PennyLaneAI/pennylane/pull/3812).

Reply:
I think that's a good start for now.  We could also consider things like `qml.expval(2* qml.PauliZ(0))`.  But for now either the current code or the code in #3812  will get the job done and be better than what we currently have.